app/models/health.rb

```diff
@@ -0,0 +1,98 @@
+class Health
+  # things to check: delayed_job, background crons, database, ECHO, CMR, URS, Opensearch, browse-scaler
+
+  def initialize
+    @ok = true
+  end
+
+  def edsc_status
+    {ok?: @ok}
+  end
+
+  def delayed_job_status
+    # The job hasn't been run yet. Go on to check retrieval.created_at. If the time is more than 10 minutes (threshold)
+    # ago, consider the delayed job is down.
+    job = DelayedJob.last
+    retrieval = Retrieval.find_by_id(job.id)
+
+    if job.run_at.nil? && retrieval.created_at < 10.minutes.ago
```

Comment:
First, I think you want retrieval.created_at > 10.minutes.ago.  But I think the way this whole condition is structured is going to cause problems. Consider that every time a new retrieval comes in, we could potentially falsely report that the jobs are running for the next 10 minutes. If we have a retrieval coming in every minute, we'll say we're processing them all, no matter what the state of the background job.

To do the grace period correctly, you need something like 
  
    DelayedJob.find('created_at > ? && run_at IS NULL', 10.minutes.ago)

If that comes back with a result, there is a stale job that's never been run.